src/main/java/software/amazon/smithy/lsp/SmithyTextDocumentService.java

```diff
@@ -283,6 +283,15 @@ public CompletableFuture<Either<List<? extends Location>, List<? extends Locatio
                         .map(Shape::getId)
                         .filter(shape -> shape.getName().equals(found))
                         .findFirst();
+                // If none of the neighbors match the token, check traits applied to members of the shape.
+                if (!target.isPresent()) {
+                    target = shapeWalker.walkShapes(initialShape).stream()
```

Comment:
Does this need to be distinct from the previous `walkShapes` call? They seem to be doing similar work with a separate `map` call if a shape is a member or not, which means they could be reconciled in to one.

Reply:
Good call. Combined in [f3cda43](https://github.com/awslabs/smithy-language-server/pull/43/commits/f3cda43be9595b7363109d6bc6b0f04e2051f5d5).